test/test_hpylong.py

```diff
@@ -3,6 +3,33 @@
 
 class TestLong(HPyTest):
 
+    def magic_int(self, v):
```

Comment:
I fear that defining classes here (outside the test functions) won't work for pypy apptests. It might be easier either to:

1. copy&paste the magic classes inside each test

2. add a single test which tests the behavior for all the various `HPyLong_As*` functions

Reply:
Are these really outside of the test functions? These methods are only called from within test functions.

Reply:
yes, but they are defined outside. PyPy's applevel tests only see the content of the single method, and the "self" that they receive is not an instance of `TestLong`, it's some fake object. And they don't get the same `globals()` also, which is the reason why you need to put `import pytest` inside the method.

There are ways to stick arbitrary objects into the app-level version of `self` (see e.g. `self.w_make_module = self.space.wrap(interp2app(descr_make_module))` inside `support.py`) but they are complex and very hard to integrate with hpy's test suite, which is supposed to run also outside pypy.

Reply:
Ah, I did not know that `self` was a fake object (despite having recently edited the bit of `support.py` which sets up the fake object!). I will make a plan.

Reply:
Good news -- I ran these in pypy as app-level tests and they just work?